src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,177 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(MyLinkedList.Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    public void removeFirst() {
+        if (size != 1) {
+            head.next.prev = null;
+            head = head.next;
+        } else {
+            head = null;
+            tail = null;
+        }
+    }
+
+    public void removeLast() {
+        tail.prev.next = null;
+        tail = tail.prev;
+    }
+
     @Override
     public void add(T value) {
+        final Node<T> l = tail;
+        final Node<T> temp = new Node<>(l, value, null);
+        if (tail == null) {
+            tail = temp;
+            head = temp;
+        } else {
+            tail.next = temp;
+            temp.prev = tail;
+            tail = temp;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Incorrect index " + index);
+        }
```

Comment:
you have checkIndex() method for that.

Reply:
here is a slightly different condition

Reply:
yes, but you can put checkIndex() after if(index == size){...}